src/components/App.js

```diff
@@ -4,13 +4,32 @@ import Display from './Display';
 import ButtonPanel from './ButtonPanel';
 import calculate from '../logic/calculate';
 
-function App() {
-  return (
-    <>
-      <Display calculate={calculate} />
-      <ButtonPanel />
-    </>
-  );
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      total: '0',
+      next: '',
+      operation: '',
```

Comment:
- [ ] Use `null` instead of  '0', ' '.  Returning null is usually the best idea if you intend to indicate that no data is available.